plugins/gatsby-transformer-home-example-code/gatsby-node.js

```diff
@@ -1,28 +1,33 @@
-const {readdirSync, readFileSync} = require('fs');
-const {join, resolve} = require('path');
+const crypto = require(`crypto`);
+
+const createContentDigest = obj =>
+  crypto
+    .createHash(`md5`)
+    .update(obj)
+    .digest(`hex`);
 
 // Store code snippets in GraphQL for the home page examples.
 // Snippets will be matched with markdown templates of the same name.
-exports.sourceNodes = ({graphql, actions}) => {
+exports.onCreateNode = async ({actions, node, loadNodeContent}) => {
   const {createNode} = actions;
+  const {absolutePath, ext, name, relativeDirectory, sourceInstanceName} = node;
 
-  const path = resolve(__dirname, '../../content/home/examples');
-  const files = readdirSync(path);
-
-  files.forEach(file => {
-    if (file.match(/\.js$/)) {
-      const code = readFileSync(join(path, file), 'utf8');
-      const id = file.replace(/\.js$/, '');
-
-      createNode({
-        id,
-        children: [],
-        parent: 'EXAMPLES',
-        internal: {
-          type: 'ExampleCode',
-          contentDigest: JSON.stringify(code),
-        },
-      });
-    }
-  });
+  if (
+    sourceInstanceName === 'content' &&
+    relativeDirectory === 'home/examples' &&
+    ext === '.js'
+  ) {
+    const code = await loadNodeContent(node);
+    createNode({
+      id: name,
+      children: [],
+      parent: 'EXAMPLES',
```

Comment:
`parent` needs set to `node.id` so Gatsby doesn't GC it when starting up Gatsby https://github.com/gatsbyjs/gatsby/blob/55539ca5c95ca3a55dd96085811758b2b3e175bd/packages/gatsby/src/utils/source-nodes.js#L57